### DISCUSSION TITLE
perl 5.38 classes

### DISCUSSION THREAD

**Discussion by s1037989** [mojolicious/mojo#2083] [General]
I really like Mojo::Base and I use it as the basis for basically every app I write, even non-Mojolicious web apps.

I'm just curious...  is there a play for perl's new `class` in Mojo in the future?  Is there any benefit to be gained by leveraging perl core `class`?

**Comment by kraih** [comment]
We should probably add a `-classes` flag to `Mojo::Base` at some point. Once we are confident the core of the feature won't change too much anymore.

**Comment by Grinnz** [comment]
There will very likely be a performance benefit to consider, though it does not support some of the encapsulation breaking that is normal with basic OO, it might be worth prototyping a wrapper on CPAN.
